test/jdk/com/sun/jdi/lib/jdb/Debuggee.java

```diff
@@ -127,33 +126,26 @@ public ProcessBuilder prepare() {
         }
 
         public Debuggee launch(String name) {
-            return new Debuggee(prepare(), name, waitForPortPrint, expectedOutputBeforeThrow);
+            return new Debuggee(prepare(), name, s -> {
```

Comment:
Address detector functions can be passed without wrapping:
`
            return new Debuggee(prepare(), name,
                                              onthrow.isEmpty() ? JDWP::parseListenAddress : Debuggee::parseLaunchEchoListenAddress)
`
But maybe current code is more readable, so it's up to you